api/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -116,6 +116,16 @@ public abstract class LoadBalancer {
   public static final Attributes.Key<Map<String, ?>> ATTR_HEALTH_CHECKING_CONFIG =
       Attributes.Key.create("internal:health-checking-config");
 
+  @Internal
+  public static final LoadBalancer.CreateSubchannelArgs.Key<LoadBalancer.SubchannelStateListener>
+      HEALTH_CONSUMER_LISTENER_ARG_KEY =
+      LoadBalancer.CreateSubchannelArgs.Key.create("internal:health-check-consumer-listener");
+
+  @Internal
+  public static final Attributes.Key<LoadBalancer.SubchannelStateListener>
+      HEALTH_PRODUCER_LISTENER_KEY =
```

Comment:
As discussed offline, nothing uses the value here. It could maybe be a Boolean. But also, it seems the HealthProducerUtil should be reading it, but it isn't.

Reply:
I updated to make it a boolean. The healthProducerUtil helper reads it.  If it is not set, it means this is the root health producer.